### DISCUSSION TITLE
Donations dApp: Experimenting with native token XML inside Soroban Smart Contracts

### DISCUSSION THREAD

**Discussion by esteblock** [stellar/sorobounty-spectacular#16] [General]
Hello! 
In order to experiment with the native token XLM inside Smart Contracts, I built a Donations dApp where there is
- an accepted **token** (we can set the native XLM) as the accepted token
- a **receiver** that later can withdraw all the accumulated tokens (in this case XML)

You can find:
- The whole code + instructions and disussions here: https://github.com/esteblock/donations-dapp-soroban
- A code interacting with this contract: https://github.com/esteblock/donations-dapp-soroban/blob/main/test_soroban_cli.sh
- A tutorial in dev.to [here](https://dev.to/esteblock/token-playground-chapter-10-write-the-native-token-xlm-contract-using-soroban-cli-1ga9)
- The same tutorial in the [Token Playground](https://token-playground.gitbook.io/guide/index/11_use_xlm_inside_contract)

Discussions and learnings:
In order to do this, I have learnt a lot of things. Also I encountered some errors, and I opened discussions in Discord:

- [First tx failing](https://discord.com/channels/897514728459468821/1145688416432963705/1145688416432963705)
- [Calling the native token inside rs-soroban-sdk tests](https://discord.com/channels/897514728459468821/1145462925109231726/1145462925109231726)
- [Create asset from string](https://discord.com/channels/897514728459468821/1145458616086843423/1145458616086843423) (**solved thanks to @vinamogit**)
- [Getting native address inside a smart contract](https://discord.com/channels/897514728459468821/1121029750912204840/1145443740249505863)

**Comment by briwylde08** [comment]
Amazing, thank you! We will review and get back to you within a week. 

Best,
Bri

**Comment by briwylde08** [comment]
Hello @esteblock we've reviewed your submission! [🎉](https://emojipedia.org/party-popper/)

Given our guidelines, your submission may qualify for an award of $550 worth of XLM. To receive that award, please fill out [this form](https://stellarform.typeform.com/to/HBjEyCJt) completely and accurately.

We can also wait for your last submission to get reviewed and do this in one big chunk. Let me know if you have any questions! 

Best,
Bri

**Comment by esteblock** [comment]
Thanks! Form filled!

**Comment by briwylde08** [comment]
Awarded per: https://github.com/stellar/sorobounty-spectacular/discussions/17
